SingularityS3Uploader/src/main/java/com/hubspot/singularity/s3uploader/SingularityS3UploaderDriver.java

```diff
@@ -209,6 +221,10 @@ public void shutdown() {
       runLock.unlock();
     }
 
+    if (fileSyncFuture != null) {
+      fileSyncFuture.cancel(true);
```

Comment:
why allow interrupts here but not for the upload future?

Reply:
Good Q, I don't remember why I put that there... Will take a second look at it

Reply:
Ah right, this one is just checking for new files, interrupting in the middle won't actually interrupt an upload. The uploaders themselves we want to give a better chance to finish